packages/lib/src/interactions/models.ts

```diff
@@ -8,12 +8,14 @@ export enum MouseButton {
 }
 
 export interface Selection {
-  startPoint: Vector2;
-  endPoint: Vector2;
-  worldStartPoint: Vector3;
-  worldEndPoint: Vector3;
+  world: Rect3;
+  data: Rect2;
 }
 
+export type Rect<T extends Vector2 | Vector3> = [T, T];
+export type Rect2 = Rect<Vector2>;
+export type Rect3 = Rect<Vector3>;
```

Comment:
The new `Rect` models I mentioned yesterday.

Reply:
It might be convenient for typing utility functions that can work with both `Rect2` and `Rect3`. None yet, though. FYI, this came useful when I tried to add a generic to the `SelectionTool` component -- the idea was that if you passed `transformSelection: (worldSelection: Rect3) => Rect2`, then you'd end up with `Rect2` in the callbacks -- it worked but wasn't quite right for consumers for various reasons.

Reply:
> It might be convenient for typing utility functions that can work with both Rect2 and Rect3.

Yeah, that was my first guess. No complains keeping it like this for now.